reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -6012,11 +6014,13 @@ public int getPrefetch() {
 	 *
 	 * @return a {@link Flux} limiting downstream's backpressure and customizing the
 	 * replenishment request amount
-	 * @see #publishOn(Scheduler, int)
+	 * @see #publishOn(Scheduler)
+	 * @see #prefetch(int, int)
 	 * @see #limitRequest(long)
 	 */
 	public final Flux<T> limitRate(int highTide, int lowTide) {
-		return onAssembly(this.publishOn(Schedulers.immediate(), true, highTide, lowTide));
+		Flux<T> prefetch = this.prefetch(highTide, lowTide);
+		return prefetch.publishOn(Schedulers.immediate(), true);
```

Comment:
I wonder if the `publishOn` is needed at all now that the prefetching logic has been moved to the `prefetch` operator. the original idea behind this operator was to re-shape requests according to lowTide/highTide and introduce limit replenishing, and it was using `publishOn` as the best approximation of that (hence the `Schedulers.immediate`).

Reply:
I just revert back to the previous behavior after removing the prefetches